File: Partner API V2/Ruby/onescanhmac.rb
require "openssl"

module Onescan

  class HMAC

    def self.encode(data, secret)
      digest = OpenSSL::Digest.new("sha1")
      return OpenSSL::HMAC.hexdigest(digest, secret, data)
    end

    # Validates that the signature in the header is the same as
    # the one created using our secret key
    def self.validateSignature(data,secret,signature)
      if signature
        hmac = Onescan::HMAC.encode(data,secret)
        if hmac == signature
          return
        end
        raise "HTTP Error: Header signature is invalid"
      else
        raise "HTTP Error: No signature found in the header"
      end
    end

  end

end

File: Partner API V2/Python/cgi-bin/onescanHMAC.py
from hashlib import sha1
import hmac

# HMac helper class
class onescanHMAC:
	@staticmethod
	def encode(data, secret):
		response=hmac.new(secret,data,sha1).hexdigest()
		return response

    # Validates that the signature in the header is the same as
    # the one created using our secret key
	@staticmethod
	def validateSignature(data,secret,signature):
		if signature:
			hmac = onescanHMAC.encode(data,secret)
			if hmac == signature:
				return
			raise Exception("HTTP Error: Header signature is invalid")

		raise Exception("HTTP Error: No signature found in the header")
